### PR TITLE
Rewrite editor lists in editor-setup.md

### LINKED ISSUE #33: editor-setup.md is too opinionated
It is saying "these editors are BAD" and listing a bunch of common editors. It's just not a good thing.

### PR DESCRIPTION
**Pull request by georgerahul24**
Are these changes ok?
I had rewritten the part where it says `bad editor` to `might not be suitable` at this stage

Fixes #33

### CODE REVIEW

**Comment by Akuli** [comment]
I think we should rewrite the editor lists. Maybe we should have 3 categories instead of 2:
- Relatively simple text editors that also run your code (VScode, Thonny, Porcupine, Geany, ...)
- IDEs and editors that are likely too advanced for a beginner (Vim, Emacs, PyCharm, ...)
- Just text editors that don't run your code (gedit, pluma, nano, notepad++, ...)

We should also mention that it doesn't matter which editor you use: everyone has their favorite editor, and there's no "right" or "wrong" choice for what editor to use.

Also, even though I am the main author of Porcupine, I don't think we should recommend it over other editors in its category. We should instead tell people to have a quick look at each editor in the list and pick their favorite.

What do you think?

**Comment by georgerahul24** [comment]
I have divided into three parts like you had hinted..Is that fine?

**Comment by Akuli** [comment]
I like it! I will review in more detail later.

File: basics/editor-setup.md [hunk 1]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
-- NetBeans
-- Notepad
-- Pluma
-- Spyder
-- Vim
-- Wingware
-
-This list doesn't contain all bad editors, but these are editors that
-people often try to use. If you know a bad editor and you think I should
+The editors can be broadly divided into three:-
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
The editors can be broadly divided into three categories:
```

File: basics/editor-setup.md [hunk 2]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+
+#### The Basic Text Editors
+These editors are usually found in-built in the OS as part of its productive suite. They do not have features like
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
These editors usually come with the operating system. They do not have features like
```

File: basics/editor-setup.md [hunk 3]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+code compilation/running, auto-completion, syntax highlighting, etc. that makes one's life easier. They are  usually used for relatively simple
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
running code, auto-completion, etc. that make programming easier. They are usually used for relatively simple
```

Deleting "syntax highlighting" because 3 of the 4 editors mentioned below have it.

File: basics/editor-setup.md [hunk 4]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+text editing and is not preferred  for coding.  
+
+A few popular ones in this category are:
+- Notepad (Windows)
+- Gedit (Linux/Ubuntu)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- Gedit (Linux)
```

File: basics/editor-setup.md [hunk 5]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+- Notepad ++ (Windows)
+- Nano (Linux,Mac OS)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- Nano (Linux/Mac OS)
```

File: basics/editor-setup.md [hunk 6]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
-- Wingware
-
-This list doesn't contain all bad editors, but these are editors that
-people often try to use. If you know a bad editor and you think I should
+The editors can be broadly divided into three:-
+
+#### The Basic Text Editors
+These editors are usually found in-built in the OS as part of its productive suite. They do not have features like
+code compilation/running, auto-completion, syntax highlighting, etc. that makes one's life easier. They are  usually used for relatively simple
+text editing and is not preferred  for coding.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
text editing. Most programmers do not use these editors for programming.
```

File: basics/editor-setup.md [hunk 7]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+text editing and is not preferred  for coding.  
+
+A few popular ones in this category are:
+- Notepad (Windows)
+- Gedit (Linux/Ubuntu)
+- Notepad ++ (Windows)
+- Nano (Linux,Mac OS)
+
+#### Advanced Text Editors
+The text editors in this category are  mainly preferred for coding at a beginner stage. They have features like auto-completion, syntax highlighting,
```
Comment (Akuli) [inline basics/editor-setup.md]: I don't think these editors are for beginners only (except maybe Thonny and IDLE).

```suggestion
The text editors in this category have features like auto-completion, syntax highlighting,
```

File: basics/editor-setup.md [hunk 8]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+code compilation, etc. They are "smart" editors have "smart" features like Error Checking, Debugging, etc. They are relatively easy to learn and have the necessary features
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
running and debugging code, highlighting errors, etc. They are relatively easy to learn and have the necessary features
```

File: basics/editor-setup.md [hunk 9]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+to start your programming journey.
+
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually is preinstalled with newer versions of Python) (Windows/Linux/Mac OS)
+- Thonny (Windows/Linux/Mac OS)
+- Porcupine(Windows/Linux/Mac OS)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- [Porcupine](https://github.com/Akuli/porcupine) (created by the author of this tutorial) (Windows/Linux/Mac OS)
```

File: basics/editor-setup.md [hunk 10]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+- Nano (Linux,Mac OS)
+
+#### Advanced Text Editors
+The text editors in this category are  mainly preferred for coding at a beginner stage. They have features like auto-completion, syntax highlighting,
+code compilation, etc. They are "smart" editors have "smart" features like Error Checking, Debugging, etc. They are relatively easy to learn and have the necessary features
+to start your programming journey.
+
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually is preinstalled with newer versions of Python) (Windows/Linux/Mac OS)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- IDLE (Usually comes with Python) (Windows/Linux/Mac OS)
```

File: basics/editor-setup.md [hunk 11]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
 editors, and most programmers have a favorite editor that they use for
```
Comment (Akuli) [inline basics/editor-setup.md:22]: A few lines above it says that IDLE won't be used in this tutorial. Please edit it to say e.g. that you can use IDLE but you should probably explore other options as well.

File: basics/editor-setup.md [hunk 12]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+The text editors in this category are  mainly preferred for coding at a beginner stage. They have features like auto-completion, syntax highlighting,
+code compilation, etc. They are "smart" editors have "smart" features like Error Checking, Debugging, etc. They are relatively easy to learn and have the necessary features
+to start your programming journey.
+
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually is preinstalled with newer versions of Python) (Windows/Linux/Mac OS)
+- Thonny (Windows/Linux/Mac OS)
+- Porcupine(Windows/Linux/Mac OS)
+- Geany(Windows/Linux/Mac OS)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- Geany (Windows/Linux/Mac OS)

**We recommend that you look into a few of these editors and install your favorite one.**
```

File: basics/editor-setup.md [hunk 13]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+
+#### Professional Level Editors / IDEs
```
Comment (Akuli) [inline basics/editor-setup.md]: Professionals use editors from the above category as well. (I use Porcupine at work more than any other editor)

```suggestion
#### IDEs
```

File: basics/editor-setup.md [hunk 14]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+code compilation/running, auto-completion, syntax highlighting, etc. that makes one's life easier. They are  usually used for relatively simple
+text editing and is not preferred  for coding.  
+
+A few popular ones in this category are:
+- Notepad (Windows)
+- Gedit (Linux/Ubuntu)
+- Notepad ++ (Windows)
+- Nano (Linux,Mac OS)
+
+#### Advanced Text Editors
```
Comment (Akuli) [inline basics/editor-setup.md]: I don't like "Advanced", it sounds like something that isn't suitable for beginners.

```suggestion
#### Smart Text Editors
```

File: basics/editor-setup.md [hunk 15]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually is preinstalled with newer versions of Python) (Windows/Linux/Mac OS)
+- Thonny (Windows/Linux/Mac OS)
+- Porcupine(Windows/Linux/Mac OS)
+- Geany(Windows/Linux/Mac OS)
+
+#### Professional Level Editors / IDEs
+This category of text editors are usually professional grade pieces of software. They are mostly proprietary and paid. They have a steep
+learning curve due to host of features they have to offer. Usually `IDE`s or `Integrated Development Environment`s fall under this.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
learning curve because of how many features they have.
```

File: basics/editor-setup.md [hunk 16]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+IDEs usually consolidates the basic tools that are necessary to not only build but also test the software. These types of editors are generally not preferred
```
Comment (Akuli) [inline basics/editor-setup.md]: Testing isn't really essential to whether something is an IDE or a text editor.

```suggestion
These types of editors are generally not preferred
```

File: basics/editor-setup.md [hunk 17]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+in the beginning stage and is usually preferred when writing complex pieces of software.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
in the beginning stage. They are meant to be used for writing complex and large pieces of software.
```

File: basics/editor-setup.md [hunk 18]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+
+A few popular ones in this category are:
```
Comment (Akuli) [inline basics/editor-setup.md:57]:
```suggestion
A few popular IDEs are:
```

File: basics/editor-setup.md [hunk 19]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+- Visual Studio (Not be confused with `Visual Studio Code`) (Windows)
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
- Visual Studio (Not be confused with *Visual Studio Code*) (Windows)
```

File: basics/editor-setup.md [hunk 20]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+- Pycharm (Windows/Linux/Mac OS)
+- Vim (Windows/Linux/Mac OS)
+- Emacs (Windows/Linux/Mac OS)
+
+There are no "right" or "wrong" editors.The preference of an editor
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
As already mentioned, there are no "right" or "wrong" editors. The preference of an editor
```

File: basics/editor-setup.md [hunk 21]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+is a personal choice and thus exploring all the options out there would be necessary to choose your preferred editor.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
is a personal choice and we recommend trying different editors.
```

File: basics/editor-setup.md [hunk 22]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+This list doesn't contain all the editors out there but a few popular ones.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
The lists on this page don't contain all editors, but just a few of the most popular ones.
```

File: basics/editor-setup.md [hunk 23]
```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
...
+If you know another editor that has not been included, and you think I should
 mention it here, please [let me know](../contact-me.md).
```
Comment (Akuli) [inline basics/editor-setup.md]: I don't think we need this anymore. The lists are pretty good right now, and the bottom of the page basically says the same thing.
```suggestion
```

File: basics/editor-setup.md [hunk 24]
```diff
@@ -79,7 +89,7 @@ echoes it back, but if you make a file that contains nothing but a
 ***
 
 If you have trouble with this tutorial, please
-[tell me about it](../contact-me.md) and I'll make this tutorial better,
+[tell me about it,](../contact-me.md) and I'll make this tutorial better,
```
Comment (Akuli) [inline basics/editor-setup.md]: If you want to fix this, please do it in a separate pull request. The right thing to change would be the `update-ends.py` script. 

```suggestion
[tell me about it](../contact-me.md) and I'll make this tutorial better,
```

**Review by Akuli — COMMENTED** [review]
As I said, I really like what you wrote overall :)

This is a big pile of small nits. I mostly simplified the grammar. I want the tutorial to be usable for people who aren't good at english, so I try to write in a simple way.

**Comment by georgerahul24** [comment]
> As I said, I really like what you wrote overall :)
> 
> This is a big pile of small nits. I mostly simplified the grammar. I want the tutorial to be usable for people who aren't good at english, so I try to write in a simple way.

Its fine...going through the suggestions I do feel that the sentences are more complex than I expected them to be.

**Comment by georgerahul24** [comment]
I hope that I resolved all the conflicts...Incase I left somethingg out it would be great if you could bring that to my notice..

File: basics/editor-setup.md [hunk 25]
```diff
@@ -13,48 +13,58 @@ like automatically displaying different things with different colors,
 but also highlighting mistakes in the code, and coloring syntax.
 
 If you are on Windows or Mac OSX you have probably noticed that your
-Python came with an editor called IDLE. We are not going to use it
-because it's lacking some important features, and most experienced
-programmers (including me) don't use it or recommend it.
+Python came with an editor called IDLE. You could start using that as your default editor 
+but, it would be preferable to find other editors that supports more features than the IDLE.
```
Comment (Akuli) [inline basics/editor-setup.md]:
```suggestion
Python came with an editor called IDLE. You can use IDLE, but we recommend exploring other options first.
```

File: basics/editor-setup.md [hunk 26]
```diff
@@ -13,48 +13,58 @@ like automatically displaying different things with different colors,
...
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually comes with Python) (Windows/Linux/Mac OS)
+- Thonny (Windows/Linux/Mac OS)
+- [Porcupine](https://github.com/Akuli/porcupine) (created by the author of this tutorial) (Windows/Linux/Mac OS)
+- Geany (Windows/Linux/Mac OS)
+
+**We recommend that you look into a few of these editors and install your favorite one.**
+
+#### IDEs
```
Comment (Akuli) [inline basics/editor-setup.md]: I imagine many people would get upset if we call Vim and Emacs "IDEs".

```suggestion
#### IDEs and advanced editors
```

**Review by Akuli — COMMENTED** [review]
The last couple tweaks...

File: basics/editor-setup.md (hunk 26 above)
Comment (georgerahul24) [reply basics/editor-setup.md]: Even I had the same dilemma on where to add but, I did not want to add a new category..This is a great workaround...

**Comment by Akuli** [comment]
Thanks!
